pkg/core/transaction/transaction.go

```diff
@@ -468,3 +469,33 @@ func (t *Transaction) ToStackItem() stackitem.Item {
 		stackitem.NewByteArray(t.Script),
 	})
 }
+
+// Copy creates a deep copy of the Transaction, including all slice fields. Cached values like
+// 'hashed' and 'size' are reset to ensure the copy can be modified independently of the original.
+func (t *Transaction) Copy() *Transaction {
+	if t == nil {
+		return nil
+	}
+	txCopy := *t
+
+	if t.Attributes != nil {
+		txCopy.Attributes = make([]Attribute, len(t.Attributes))
+		copy(txCopy.Attributes, t.Attributes)
+	}
+	if t.Signers != nil {
+		txCopy.Signers = make([]Signer, len(t.Signers))
+		copy(txCopy.Signers, t.Signers)
+	}
+	if t.Scripts != nil {
+		txCopy.Scripts = make([]Witness, len(t.Scripts))
+		for i, script := range t.Scripts {
+			txCopy.Scripts[i] = *script.Copy()
+		}
+	}
+	txCopy.Script = bytes.Clone(t.Script)
+
+	txCopy.hashed = false
+	txCopy.size = 0
```

Comment:
Technically, it's enough, but I'd prefer to keep the resulting transaction consistent, and thus, `hash` should also be reset.